app/models/photo.rb

```diff
@@ -141,6 +141,6 @@ def self.visible(current_user, person, limit=:all, max_time=nil)
              else
                Photo.where(author_id: person.id, public: true)
              end
-    photos.order("created_at desc")
+    photos.where(pending: false).order("created_at DESC")
```

Comment:
I thought about leaving out the `.where(pending: false)` here. That would make the difference, that the user can see his own pending photos (and can delete them). Other users can't see pending photos anyway, because they are default private and have no visibilities.